--- newlib-4.1.0/newlib/configure.host	2020-05-20 05:05:39.039807077 +0000
+++ newlib-4.1.0/newlib/configure.host	2020-05-20 05:05:32.435780002 +0000
@@ -324,6 +324,7 @@
 	;;
   sh | sh64)
 	machine_dir=sh
+	newlib_cflags="${newlib_cflags} -DREENTRANT_SYSCALLS_PROVIDED -ffunction-sections -fdata-sections"
 	;;
   sparc*)
 	machine_dir=sparc
--- newlib-4.1.0/newlib/libc/sys/sh/syscalls.c	2020-01-22 10:05:51.000000000 +0000
+++ newlib-4.1.0/newlib/libc/sys/sh/syscalls.c	2020-05-21 19:54:46.620029245 +0000
@@ -1,3 +1,5 @@
+// Disable syscalls, Naomi does not have them.
+#if 0
 #include <_ansi.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -226,3 +228,4 @@
   else
     return __setup_argv_for_main (argc);
 }
+#endif
